main.c

```diff
@@ -593,6 +609,8 @@ int main (int argc, char **argv) {
 		if (sock == -1)
 			fprintf(stderr, "could not create socket\n");
   
+		// make sure the structure is in a known stat before caling connect()
+		memset(&server,0,sizeof(server));
```

Comment:
Think this is not necessary, as the struct will be initialized in the following lines ....

Reply:
```
sizeof(server) : 16
sizeof(server.sin_addr.s_addr) : 4
sizeof(server.sin_family) : 2
sizeof(server.sin_port) : 2
```

8 of 16 bytes initialized. The rest remains uninitialized.
Please see the documentation on connect()
https://www.ibm.com/support/knowledgecenter/SSLTBW_2.2.0/com.ibm.zos.v2r2.bpxbd00/connect.htm
Search for "Connect to server bound to a specific interface".